### PR TITLE
Urls and redirects

### PR DESCRIPTION
**Pull request by krulis-martin**

### CODE REVIEW

File: src/pages/ChangePassword/ChangePassword.js [hunk 1]
```diff
@@ -40,15 +40,15 @@ class ChangePassword extends Component {
...
     if (canUseDOM) {
-      const hash = window.location.hash;
-      if (hash.length === 0) {
+      const search = window.location.search;
+      if (search.length === 0) {
         const {
           push,
           links: { RESET_PASSWORD_URI },
         } = this.props;
         push(RESET_PASSWORD_URI); // no hash -> redirect to the reset form
```
Comment (SemaiCZE) [inline src/pages/ChangePassword/ChangePassword.js]: comment says `no hash` which is now not true. But it is just a minor issue

File: src/redux/middleware/apiMiddleware.js [hunk 2]
```diff
@@ -40,6 +41,11 @@ export const apiCall = (
   meta: { endpoint, ...meta },
 });
 
+const lastLocation = {
```
Comment (SemaiCZE) [inline src/redux/middleware/apiMiddleware.js]: What is this for?

Comment (krulis-martin) [reply src/redux/middleware/apiMiddleware.js]: Well, there is a mechanism that aborts all pending fetch requests, if the location is changed. Unfortunately, multiple LOCATION_CHANGED actions may be dispatched with the same URL (so not all of them change the location). We need to remember last location, so we can determine, whether the location is actually being changed when the action is dispatched.

Comment (SemaiCZE) [reply src/redux/middleware/apiMiddleware.js]: ah, thanks!

**Pull request merged**